### PR TITLE
FR-14304 - Trigger Oauth pipeline after publish

### PR DESCRIPTION
**Pull request by rotemzif1**
This PR introduces a new workflow for triggering oauth service pipeline of opening a PR that updating frontegg, on our publish flow. (Oauth workflow - https://github.com/frontegg/oauth-service/actions/workflows/update-react-dependency.yaml)

This PR includes 4 files:
1. Publish.yaml - the actual necessary flow which was fully tested
2. pre-release - copy-paste of the workflow just for testing on an alpha to make sure that also waiting for npm registery is working ok. It will be removed right after.
3. push.yaml - Guy asked me to change the version of the scripts
4. waiting for npm registry - code that was copied from admin box, to make sure that npm has been done publishing the new version before we are trying to bump it in oauth service.

This PR will be pushed the same way to trigger dashboard service as well.

### CODE REVIEW

**Review by yuvalotem1 — APPROVED** [review]
Amazing!

File: .github/scripts/wait-for-npm-indexing.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        const result = JSON.parse(execSync(`npm show ${packages[i]} --json dist-tags`).toString('utf8').trim());
+        allPackagesIndexed = allPackagesIndexed && (result.alpha === version || result.latest === version);
+      } catch (e) {
+        console.error('Failed to check version for', packages[i], e);
+        allPackagesIndexed = false;
+      }
+    }
+
+    if (!allPackagesIndexed) {
+      timeout--;
```
Comment (doregg) [inline .github/scripts/wait-for-npm-indexing.js:27]: I think we can improve the timeout logic. I think we can define how many time to run it regardless of the timeout var

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -99,6 +99,34 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           NODE_AUTH_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+      - name : Wait until NPM registry finished indexing the new version
```
[the remark this answers is not in the document]
Comment (rotemzif1) [reply .github/workflows/publish.yml:102]: Its a great idea, if we go with this direction I suggest we think together how to do it and provide it also in admin box, this is the same logic when we publish and then do something that is holding up until the version is live

[the remark this answers is not in the document]
Comment (doregg) [reply .github/workflows/publish.yml:102]: can u add a ticket? in the pipeline effort epic

**Pull request merged**